[IMP] accounting update foreign currency (#18319)
odoo/documentation: content/applications/finance/accounting/bank/foreign_currency.rst

===========================================

In Odoo, every transaction is recorded in the default currency of the company, and reports are all
based on that default currency. With a bank account in a foreign currency, for every
transaction, Odoo stores two values:

-  The debit/credit in the currency of the company.
-  The debit/credit in the currency of the bank account.

Currency rates are updated automatically using a banking institution's web services. By default,
Odoo uses the European Central Bank's web services but other options are available.

> Configuration

Reviewer: Just a little reminder to add custom anchors (with the structure 'app name/rst file name/section name') to all headings so they can easily be referenced when needed. Here's where it's referred to in the guidelines if that helps: https://www.odoo.com/documentation/19.0/contributing/documentation/rst_guidelines.html#custom-anchor

=============

Configure currencies
--------------------

> To work with multiple currencies, go to :menuselection:`Accounting --> Configuration --> currencies`
> then, :doc:`activate and configure new currencies <../get_started/multi_currency>`.

Reviewer: ```suggestion
To work with multiple currencies, go to :menuselection:`Accounting --> Configuration --> Currencies`,
then :doc:`activate and configure new currencies <../get_started/multi_currency>`.
```

> Set up manual or automatic :ref:`currency rate <multi-currency/config-rates>` and, define the
> exchange difference entries accounts and journals to record gains or losses incurred.

Reviewer: ```suggestion
Set up manual or automatic :ref:`currency rate <multi-currency/config-rates>` updates and :ref:`define the
accounts and journals used to record gains or losses incurred on :ref:`exchange rate differences <multi-currency/config-exch-diff>`.
```

- added '[currency rate] updates' since we're talking about how the currency rate is _updated_
- I found '... the exchange difference entries accounts and journals ...' a bit hard to read, hence my suggestion above. I also think it's worth linking to `<multi-currency/config-exch-diff>`

.. seealso::
   :doc:`Multi-currency system <../get_started/multi_currency>`

> Create a new bank account

Reviewer: I'm a little confused about this ... this isn't about _creating_ a bank account per se, right? (Like, not in the sense of 'Create a bank account' on [this page(https://www.odoo.com/documentation/19.0/applications/finance/accounting/bank.html#manage-bank-and-cash-accounts))

It's more about adding or connecting a bank account?
Maybe you can explain this to me so I understand better, and can better suggest how to phrase it.

the option to choose the :guilabel:`Service` you wish to obtain currency rates from.

Click on the Update now button (:guilabel:`🗘`) besides the :guilabel:`Next Run` field to update
the currency rates manually.
.. seealso::
   :doc:`Multi-currency system <../get_started/multi_currency>`

Create a new bank account
-------------------------

In the accounting application, go to :menuselection:`Accounting --> Configuration --> Journals` and
create a new one. Enter a :guilabel:`Journal Name` and set the :guilabel:`Type` to `Bank`. In the
:guilabel:`Journal Entries` tab, enter a **short code**, a **currency**, and then finally click on
the :guilabel:`Bank Account` field to create a new account. In the pop-up window of the account
creation, enter a name, a code (ex.: 550007), set its type to `Bank and Cash`, set a currency type,
and save. When you are back on the **journal**, click on the :guilabel:`Account Number` field, and
in the pop-up window, fill out the :guilabel:`Account Number`, :guilabel:`Bank` of your account, and
save.

.. image:: foreign_currency/foreign-journal.png
   :align: center

>    :alt: Example of a created bank journal.

Reviewer: Remember to delete this one in PyCharm as it's not used any more

Upon creation of the journal, Odoo automatically links the bank account to the journal. It can be
found under :menuselection:`Accounting --> Configuration --> Accounting: Chart of Accounts`.

Vendor bill in a foreign currency
=================================

To pay a bill in a foreign currency, simply select the currency next to the :guilabel:`Journal`
field and register the payment. Odoo automatically creates and posts the foreign **exchange gain or
loss** as a new journal entry.

.. image:: foreign_currency/foreign-bill-currency.png
   :align: center

>    :alt: How to set a bill currency.

Reviewer: ... and delete this one too

> To create a new bank account, first create a bank journal via
> :menuselection:`Accounting --> Configuration --> Journals`, then add a
> :ref:`bank account <accounting/journals/bank-cash-cc>` in the journal, and configure it.

Reviewer: ```suggestion
To create a new bank account in a foreign currency, first create a bank journal via
:menuselection:`Accounting --> Configuration --> Journals`, then add a
:ref:`bank account <accounting/journals/bank-cash-cc>` in the journal, and configure it.
```

- Given what this page is about, maybe better to just be a little more specific?
- I think it'd be better to link directly to <accounting/journals/bank> rather than the section heading, since if you arrive at the higher level, it's a bit of a journey down to the relevant content

>    - :ref:`manage bank and cash accounts <accounting/bank/manage>`

Reviewer: ```suggestion
   - :ref:`Manage bank and cash accounts <accounting/bank/manage>`
```

> Unrealized Currencies Report

Reviewer: ```suggestion
Unrealized Currencies report
```

Since 'report' isn't part of the name of the report, I'd keep it lower case.
I saw this is how Dallas did for the [Audit Trail report](https://www.odoo.com/documentation/19.0/applications/finance/accounting/reporting.html#audit-trail) that was added back in June.

> This report gives an overview of all unrealized amounts in a foreign currency on the balance sheet,
> and allows the adjustment of an entry or manually set an exchange rate. To access this report, go to

Reviewer: ```suggestion
This report gives an overview of all unrealized amounts in a foreign currency on the balance sheet,
and allows entries to be adjusted or an exchange rate to be set manually. To access the report, go to
```

- Due to the edits, the first sentence ended up with a phrase that wasn't grammatically correct ('...allows the adjustment of an entry or manually set an exchange rate.'), so see my suggestion above.
- In this sentence, you mention adjustments first and then manually setting an exchange rate, while in the text you cover them in the opposite order. To guide the reader better, I'd keep the same order in both places ... maybe one makes more sense to have first? Either way, just pick and order and use it in both places.

> :menuselection:`Review --> Unrealized Currencies` and have access to all open entries in the
> balance sheet.

Reviewer: ```suggestion
:menuselection:`Accounting --> Review --> Unrealized Currencies`.
```

- Just like in the rest of the page, add the app name at the start of the menu items.
(https://www.odoo.com/documentation/19.0/contributing/documentation/rst_guidelines.html#menu-selection)
- The 'have access' doesn't work in English I'm afraid. And actually, in the intro sentence, you already explain what is shown in this report, so I think you can just delete that last part.

>    :alt: View of the Unrealized Gains/Losses journal.

Reviewer: I know the originals were center aligned, and you probably only updated the image files, but remember to remove the `:align: center` as we generally left align images in the doc (https://www.odoo.com/documentation/19.0/contributing/documentation/rst_guidelines.html#images)

> To use a different currency rate than the one set in
> :menuselection:`Accounting --> Configuration --> Settings`, click the :guilabel:`Exchange Rates`

Reviewer: Do you mean: 

```suggestion
To use a different currency rate than the one set in
:menuselection:`Accounting --> Configuration --> Currencies`, click the :guilabel:`Exchange Rates`
```

> When manually changing exchange rates, a yellow banner appears allowing a reset back to Odoo's rate.
> To do so, simply click :guilabel:`Reset to Odoo's Rate`.

Reviewer: ```suggestion
When manually changing exchange rates, a yellow banner appears allowing the rate to be reset to Odoo's rate. To do so, simply click :guilabel:`Reset to Odoo's Rate`.
```

I would just use the passive here, as it sounds more natural than 'allowing a reset'. Also, 'reset' implies 'back to', so I think that's unnecessary.

>    :alt: Banner to reset back to Odoo's rates.

Reviewer: I'm not 100% convinced we need these two images. The first image of the report (which I think is worth keeping ... but which Audrey might have vetoed 😉 ) alraedy shows the 'Exchange Rates' button, and I don't think showing it in action is _that_ valuable.

As for the banner, I also don't think it warrants having an image as it's something straightforward (i.e., no confusion to clear up).

> In order to update the balance sheet with the amount of the :guilabel:`Adjustment` column,
> click on the :guilabel:`Adjustment Entry` button in the top-left-corner. In the pop-up window,

Reviewer: ```suggestion
To update the balance sheet with the amount of the :guilabel:`Adjustment` column,
click on the :guilabel:`Adjustment Entry` button in the top-left corner. In the pop-up window,
```

> calculate and process the unrealized gains and losses.

Reviewer: Would 'unrealized gains _and/or_ losses' be better? As I guess it might be only one or the other, or both.

> gains and losses have been adjusted.

Reviewer: Same question as earlier ... would 'unrealized gains and/or losses' be better?

>    :align: center

Reviewer: I do get why you feel it's nice to show this, but ... I don't think we need it 🙈

===========================================

In Odoo, every transaction is recorded in the default currency of the company, and reports are all
based on that default currency. With a bank account in a foreign currency, for every
transaction, Odoo stores two values:

-  The debit/credit in the currency of the company.
-  The debit/credit in the currency of the bank account.

Currency rates are updated automatically using a banking institution's web services. By default,
Odoo uses the European Central Bank's web services but other options are available.

.. _accounting/foreign_currency/config:

Configuration
=============

.. _accounting/foreign_currency/config-currencies:

Configure currencies
--------------------

To work with multiple currencies, go to :menuselection:`Accounting --> Configuration --> Currencies`

> then, :doc:`activate and configure new currencies <../get_started/multi_currency>`.

Reviewer 2: ```suggestion
To work with multiple currencies, go to :menuselection:`Accounting --> Configuration --> Currencies`,
then :doc:`activate and configure new currencies <../get_started/multi_currency>`.
```

Set up manual or automatic :ref:`currency rate <multi-currency/config-rates>` updates and define the
accounts and journals used to record gains or losses incurred on :ref:`exchange rate differences
<multi-currency/config-exch-diff>`.

.. seealso::
   :doc:`Multi-currency system <../get_started/multi_currency>`

.. _accounting/foreign_currency/create-bank-journal-and-account:

Create bank journals and accounts in foreign currencies
-------------------------------------------------------

To a bank account in a foreign currency, first create a new bank journal via

> :menuselection:`Accounting --> Configuration --> Journals`.Set the journal's currency to the

Reviewer 2: ```suggestion
:menuselection:`Accounting --> Configuration --> Journals`. Set the journal's currency to the
```

foreign currency, then add the relevant :ref:`bank account <accounting/journals/bank>` in the
journal, and configure it.
If necessary (such as in the case of creating batch payment files), add a
:ref:`bank <accounting/batch/create-bank>` and a
:ref:`bank account <accounting/batch/create-bank-account>`.

.. seealso::
   - :doc:`Journals <../get_started/journals>`
   - :ref:`Manage bank and cash accounts <accounting/bank/manage>`

.. _accounting/foreign_currency/report:

Unrealized Currencies report
============================

> This report gives an overview of all unrealized amounts in a foreign currency on the balance sheet,

Reviewer 2: ```suggestion
The :guilabel:`Unrealizede Currencies` report gives an overview of all unrealized amounts in a foreign currency on the balance sheet,
```

automatic :ref:`currency rate <multi-currency/config-rates>` updates and define the accounts and
journals used to record gains or losses incurred on
:ref:`exchange rate differences <multi-currency/config-exch-diff>`.

.. seealso::
   :doc:`Multi-currency system <../get_started/multi_currency>`

.. _accounting/foreign_currency/create-bank-journal-and-account:

Create bank journals and accounts in foreign currencies
-------------------------------------------------------

To set up a bank account in a foreign currency, first create a new bank journal via
:menuselection:`Accounting --> Configuration --> Journals`. Set the journal's currency to the
foreign currency, then add the relevant :ref:`bank account <accounting/journals/bank>` in the
journal, and configure it.
If necessary (such as in the case of creating batch payment files), add a
:ref:`bank <accounting/batch/create-bank>` and a
:ref:`bank account <accounting/batch/create-bank-account>`.

.. seealso::
   - :doc:`Journals <../get_started/journals>`
   - :ref:`Manage bank and cash accounts <accounting/bank/manage>`

.. _accounting/foreign_currency/report:

Unrealized Currencies report
============================

> The :guilabel:`Unrealizede Currencies` report gives an overview of all unrealized amounts in a

Reviewer 2: ```suggestion
The :guilabel:`Unrealized Currencies` report gives an overview of all unrealized amounts in a
```
Petit typo (sorry, I think that was my fault when I made the suggestion!) but I'll go ahead and approve :)

Author: (aah but mine too, I probably read to fast and missed it :see_no_evil: )
